### PR TITLE
fix: Ensure single instance using Gtk.Application

### PR DESCRIPTION
**Pull request by SoumyaRanjanPatnaik**
The wayland session allows ilia to be spawned multiple times, causing stacked instances that need to be closed manually. This change refactors the main function to use Gtk.Application, which inherently ensures that only a single instance of the application is allowed to run. It also cleans up the Main.vala file by moving the all th application startup related logic to Application.vala.

To reproduce the issue, you can open multiple instances of ilia by holding down Super and pressing the space key more than once on the sway session without releasing Super.

### CODE REVIEW

**Comment by SoumyaRanjanPatnaik** [comment]
Tested this change in both wayland and x11 sessions and can verify it to be working.

File: src/Application.vala [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+            return base.local_command_line (ref args, out exit_status);
+        }
+
+        protected override void activate () {
+            var window = new Ilia.DialogWindow (this.arg_map);
+            window.set_application (this);
+
+            // Grab inputs from wayland backend before showing window
+            if (IS_SESSION_WAYLAND) {
+                GtkLayerShell.init_for_window (window);
```
Comment (kgilmer) [inline src/Application.vala:28]: note: it's interesting that these methods are apparently infallible.  Is an exception thrown or something in the case of some error (memcpy, etc)

File: src/Application.vala [hunk 2]
```diff
@@ -0,0 +1,202 @@
+using Gtk;
+
+namespace Ilia {
+    class Application : Gtk.Application {
```
Comment (kgilmer) [inline src/Application.vala:4]: Adding this is something I had always planned to do but never got to.  Nice work!

**Review by kgilmer — APPROVED** [review]
LGTM

File: src/Application.vala (hunk 2 above)
Comment (SoumyaRanjanPatnaik) [reply src/Application.vala:4]: Hehe :)

File: src/Application.vala (hunk 1 above)
Comment (SoumyaRanjanPatnaik) [reply src/Application.vala:28]: I'm actually not sure if the are. However, there's no clear indication in the documentation for this library about any exceptions. Maybe they are already handled?

Comment (kgilmer) [reply src/Application.vala:33]: Maybe!  Yeah the gtk/gdk docs coverage has always been spotty.

**Review by kgilmer — APPROVED** [review]
LGTM

**Pull request merged**